system/Helpers/text_helper.php

```diff
@@ -672,3 +672,26 @@ function excerpt(string $text, ?string $phrase = null, int $radius = 100, string
         return str_replace('  ', ' ', $ellPre . $prev . $phrase . $post . $ellipsis);
     }
 }
+
+if (!function_exists('str_contains')) {
+    /**
+     * Determine if a given string contains in a given substring.
+     *
+     * @param  string  $haystack
+     * @param  string|string[]  $needles
+     * @return bool
+     */
+    function str_contains(string $haystack, string $needle): bool
+    {
+        if (phpversion() >= 8) {
+            return str_contains($haystack, $needle);
+        }
+
+        if ($haystack !== '' && $needle !== '' && mb_strpos($haystack, $needle) !== false) {
```

Comment:
What's wrong? `mbstring` is a requirement for CI4.

Reply:
Since this is a polyfill of the PHP 8 function, it should work the same with that. And the native function does not depend on the `mbstring` extension.

Reply:
I see. In this case, it seems `strpos()` is better. Because `str_contains()` in PHP8 does not support multibyte chars correctly (But at least It works with UTF-8).